src/python/crash_analysis/stack_parsing/stack_analyzer.py

```diff
@@ -1199,789 +48,56 @@ def get_crash_data(crash_data,
-      if not downcast_match:
-        downcast_match = CFI_INVALID_DOWNCAST_REGEX.match(line)
-      if downcast_match:
-        state.crash_state += ' from %s' % (downcast_match.group(1)).strip("'")
-        state.found_bad_cast_crash_end_marker = True
-
-      if (UBSAN_VPTR_INVALID_VPTR_REGEX.match(line) or
-          CFI_INVALID_VPTR_REGEX.match(line)):
-        state.crash_state += ' from invalid vptr'
-        state.found_bad_cast_crash_end_marker = True
-
-      if CFI_FUNC_DEFINED_HERE_REGEX.match(line):
-        state.found_bad_cast_crash_end_marker = True
-
-      # Ubsan's -fsanitize=vptr crash extra info for member access.
-      invalid_offset_match = UBSAN_VPTR_INVALID_OFFSET_REGEX.match(line)
-      if invalid_offset_match:
-        state.crash_state += (' from base class subobject at offset %s' %
-                              invalid_offset_match.group(1))
-        state.found_bad_cast_crash_end_marker = True
-
-      if state.found_bad_cast_crash_end_marker:
-        state.crash_state += '\n'
-        state.frame_count += 1
-
-    # CFI bad-cast crash.
-    if not state.crash_type:
-      cfi_bad_cast_match = update_state_on_match(
-          CFI_ERROR_REGEX, line, state, new_type='Bad-cast', new_frame_count=0)
-      if cfi_bad_cast_match:
-        state.crash_state = 'Bad-cast to %s' % (
-            cfi_bad_cast_match.group(2).strip("'"))
-        if cfi_bad_cast_match.group(4):
-          state.crash_address = cfi_bad_cast_match.group(4)
-        state.found_bad_cast_crash_end_marker = False
-
-    # CFI bad-cast crash without extra debugging information.
-    if not state.crash_type:
-      update_state_on_match(
-          CFI_NODEBUG_ERROR_MARKER_REGEX,
-          line,
-          state,
-          new_type='Bad-cast',
-          new_frame_count=0)
-
-    # Other UndefinedBehavior Sanitizer crash.
-    ubsan_runtime_match = UBSAN_RUNTIME_ERROR_REGEX.match(line)
-    if ubsan_runtime_match and not state.crash_type and not ubsan_disabled:
-      reason = ubsan_runtime_match.group(2)
-      state.crash_type = 'UNKNOWN'
-
-      for ubsan_crash_regex, ubsan_crash_type in UBSAN_CRASH_TYPES_MAP:
-        if update_state_on_match(
-            ubsan_crash_regex, reason, state, new_type=ubsan_crash_type):
-          break
-
-      if state.crash_type == 'UNKNOWN':
-        logs.log_error(
-            'Unknown UBSan crash type: {reason}'.format(reason=reason))
-
-      state.crash_address = ''
-      state.crash_state = ''
-      state.frame_count = 0
-
-    # AddressSanitizer for memory overlap crash.
-    update_state_on_match(
-        ASAN_MEMCPY_OVERLAP_REGEX,
-        line,
-        state,
-        new_type='Memcpy-param-overlap',
-        reset=True,
-        address_from_group=2)
+  stack_parser = stacktraces.StackParser(
+      symbolized=symbolize_flag or already_symbolized,
+      detect_ooms_and_hangs=detect_ooms_and_hangs,
+      detect_v8_runtime_errors=detect_v8_runtime_errors,
```

Comment:
This detect_v8_runtime_errors is very chromium specific, wonder if we should keep it just inside stackparser as a hack.

Reply:
Yes thought about it, but it needs a a bit more refactoring to make it work. Will keep this in mind for future work here.